### PR TITLE
fix(esx_lscustom/client/main.lua): fixed lscustom glitch

### LINKED ISSUE #289: bug (esx_lscustom): free tuning glitch
Free tuning

1. Select desired change on vehicle
2. Disconnect
3. Free tuning

https://streamable.com/dfgsy3

### PR DESCRIPTION
**Pull request by AmIrReZa386**
Hi,
I fix #289 bug
I add one table and some events for save first props of vehicle
if player disconnect reset that vehicle

### CODE REVIEW

**Comment by Benzo00** [comment]
Please take a min to read this https://github.com/esx-framework/esx-legacy/issues/343

**Comment by AmIrReZa386** [comment]
> Please take a min to read this #343

thanks i edit 💖

File: [esx_addons]/esx_lscustom/server/main.lua [hunk 1]
```diff
@@ -33,6 +34,20 @@ AddEventHandler('esx_lscustom:buyMod', function(price)
 	end
 end)
 
+RegisterNetEvent('esx_lscustom:changeVehicleData')
+AddEventHandler('esx_lscustom:changeVehicleData', function(type, data)
+	local source = source
+	if type == "add" then
+		VehicleData[source] = {plate = data.plate, props = data.props}
+	elseif type == "remove" then
+		for index, value in ipairs(VehicleData) do
```
Comment (mahanmoulaei) [inline [esx_addons]/esx_lscustom/server/main.lua:43]: You can replace the loop with :

```lua
if VehicleData[source] then VehicleData[source] = nil end
```

File: [esx_addons]/esx_lscustom/server/main.lua [hunk 2]
```diff
@@ -56,4 +71,15 @@ ESX.RegisterServerCallback('esx_lscustom:getVehiclesPrices', function(source, cb
...
 	end
 	cb(Vehicles)
-end)
\ No newline at end of file
+end)
+
+AddEventHandler('esx:playerDropped', function(playerId, reason)
+	if VehicleData[playerId] then
+		TriggerClientEvent('esx_lscustom:resetCustom', playerId, VehicleData[playerId].props)
+		for index, value in ipairs(VehicleData) do
```
Comment (mahanmoulaei) [inline [esx_addons]/esx_lscustom/server/main.lua:79]: You can replace the loop with :

```lua
VehicleData[playerId] = nil
```

File: [esx_addons]/esx_lscustom/server/main.lua [hunk 3]
```diff
@@ -33,6 +34,20 @@ AddEventHandler('esx_lscustom:buyMod', function(price)
 	end
 end)
 
+RegisterNetEvent('esx_lscustom:changeVehicleData')
+AddEventHandler('esx_lscustom:changeVehicleData', function(type, data)
+	local source = source
+	if type == "add" then
+		VehicleData[source] = {plate = data.plate, props = data.props}
```
Comment (mahanmoulaei) [inline [esx_addons]/esx_lscustom/server/main.lua:41]: You can replace this line with :

```lua
VehicleData[source] = data
```

File: [esx_addons]/esx_lscustom/client/main.lua [hunk 4]
```diff
@@ -1,542 +1,434 @@
...
+						Sleep = 0
+						if not lsMenuIsShowed then
+							ESX.ShowHelpNotification(v.Hint)
+							if IsControlJustReleased(0, 38) then
+								lsMenuIsShowed = true
+
+								local vehicle = GetVehiclePedIsIn(playerPed, false)
+								FreezeEntityPosition(vehicle, true)
+								myCar = ESX.Game.GetVehicleProperties(vehicle)
+								TriggerServerEvent('esx_lscustom:changeVehicleData', "add", {plate = GetVehicleNumberPlateText(vehicle), props = myCar})
```
Comment (mahanmoulaei) [inline [esx_addons]/esx_lscustom/client/main.lua:404]: You can replace this line with :

```lua
TriggerServerEvent('esx_lscustom:changeVehicleData', "add", myCar)
```

File: [esx_addons]/esx_lscustom/client/main.lua [hunk 5]
```diff
@@ -1,542 +1,434 @@
...
+		TriggerEvent('esx_lscustom:cancelInstallMod')
+		
+		local playerPed = PlayerPedId()
+		local vehicle = GetVehiclePedIsIn(playerPed, false)
+		SetVehicleDoorsShut(vehicle, false)
+
+		if parent == nil then
+			lsMenuIsShowed = false
+			local vehicle = GetVehiclePedIsIn(PlayerPedId(), false)
+			TriggerClientEvent('esx_lscustom:changeVehicleData', "remove", {plate = GetVehicleNumberPlateText(vehicle)})
```
Comment (mahanmoulaei) [inline [esx_addons]/esx_lscustom/client/main.lua:111]: ⚠ You must trigger the server, not the client!
-----------------------------------------------
You can replace this line with :

```lua
TriggerServerEvent('esx_lscustom:changeVehicleData', "remove")
```

**Review by mahanmoulaei — COMMENTED** [review]
Haven't tested your pull request myself, but I showed you the changes you can make to avoid the unneeded loops on the server.

File: [esx_addons]/esx_lscustom/server/main.lua (hunk 1 above)
Comment (AmIrReZa386) [reply [esx_addons]/esx_lscustom/server/main.lua:43]: HI, @mahanmoulaei
Maybe he/she custom multi cars
Again, a specific data must be deleted

File: [esx_addons]/esx_lscustom/server/main.lua (hunk 2 above)
Comment (AmIrReZa386) [reply [esx_addons]/esx_lscustom/server/main.lua:79]: Again maybe use multi vehicles

File: [esx_addons]/esx_lscustom/server/main.lua (hunk 3 above)
Comment (AmIrReZa386) [reply [esx_addons]/esx_lscustom/server/main.lua:41]: yes this is true
we can get plate from vehicle's prop
but it's very easy for find vehicle in tables

File: [esx_addons]/esx_lscustom/client/main.lua (hunk 4 above)
Comment (AmIrReZa386) [reply [esx_addons]/esx_lscustom/client/main.lua:404]: Some people use space plate in esx_vehicleshop, in code we need format that to send it to server side
```
Config.PlateLetters  = 3
Config.PlateNumbers  = 3
Config.PlateUseSpace = true
```

File: [esx_addons]/esx_lscustom/client/main.lua (hunk 5 above)
Comment (AmIrReZa386) [reply [esx_addons]/esx_lscustom/client/main.lua:111]: as
```
TriggerServerEvent('esx_lscustom:changeVehicleData', "add", myCar)
```

**Pull request closed without merging**